packages/react-dom/src/client/ReactDOMComponent.js

```diff
@@ -875,7 +874,6 @@ export function diffHydratedProperties(
   let extraAttributeNames: Set<string>;
 
   if (__DEV__) {
-    suppressHydrationWarning = rawProps[SUPPRESS_HYDRATION_WARNING] === true;
```

Comment:
Moved inline since we don't want to hit the property checks for every props object.